integreat_cms/cms/linklists.py

```diff
@@ -22,6 +30,19 @@ def filter_callable(cls, objects):
         :return: Objects that passed the filter
         :rtype: ~django.db.models.query.QuerySet
         """
+        # Because of large overhead, we only want this filter during the findlinks management command.
+        if settings.LINKCHECK_COMMAND_RUNNING:
+            # Exclude links in inactive languages
+            objects = objects.annotate(
+                active=Exists(
+                    LanguageTreeNode.objects.filter(
+                        region=OuterRef(f"{objects.model.foreign_field()}__region"),
+                        language=OuterRef("language"),
+                        active=True,
+                    )
+                )
+            ).filter(active=True)
```

Comment:
Probably we need another intermediate class, because `PageTranslationLinklist` does not inherit from the existing `ContentTranslationLinklist` (this is because the `__archived` filter does nor work for pages...)

How about the following structure:
- Base class `ActiveLanguageLinklist(Linklist)` with the new `filter_callable` to exclude inactive languages
- Page link list `PageLinklist(ActiveLanguageLinklist)` inheriting the language filter
- Intermediate base class `NonArchivedLinklist(ActiveLanguageLinklist)` with the archived filter as base for events and pois
- Link lists `EventLinklist(NonArchivedLinklist)` and `POILinklist(NonArchivedLinklist)` inheriting the intermediate class

Also feel free to suggest a different naming of the classes if you have better ideas :laughing: